### PR TITLE
Add checklist for Spring Data JPA

### PR DESCRIPTION
**Pull request by dskarzh**

### CODE REVIEW

File: java-spring-boot/spring-data-jpa/checklist.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+## Do not create custom queries, use JPA methods
+
+If you are trying to write custom query using `@Query` annotation, think how you can do the same using JPA method naming.
+
+## REST endpoint naming
+
+Follow best practices when choosing names for your endpoints.
+Take a look at this article: https://nordicapis.com/10-best-practices-for-naming-api-endpoints/
+
+## Use abstract types for mappers
```
Comment (Nick97-git) [inline java-spring-boot/spring-data-jpa/checklist.md:33]: add bad and good examples

File: java-spring-boot/spring-data-jpa/checklist.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+4. `@ToString`
+5. `@EqualsAndHashcode`
+6. `@NoArgsConstructor`
+7. `@RequiredArgsConstructor`
+9. `@AllArgsConstructor`
+
+**BUT**, it should be used with caution, because it can cause performance issues. 
+Consider these articles: 
+1. https://habr.com/ru/company/haulmont/blog/564682/
+2. https://thorben-janssen.com/lombok-hibernate-how-to-avoid-common-pitfalls
```
Comment (Nick97-git) [inline java-spring-boot/spring-data-jpa/checklist.md:22]: I wouldn't add it to the checklist, because using Lombok is more like advice, it's not a mistake

Comment (dSkarzh) [reply java-spring-boot/spring-data-jpa/checklist.md:22]: Would it be better if I rewrite this paragraph, so it's more clear that Lombok is optional and this is some tips if you decide to use it?

File: java-spring-boot/spring-data-jpa/jv-springboot-data-jpa.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+            }
+    
+         Good: 
+            @RestController
+            public class ProductController {     
+                private final DtoMapper<Product, ProductRequestDto, ProductResponceDto> productMapper;
+                        
+            }
+    ```
```
Comment (hronograf) [inline java-spring-boot/spring-data-jpa/jv-springboot-data-jpa.md:28]: image
I can see ``` in preview

File: java-spring-boot/spring-data-jpa/jv-springboot-data-jpa.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+5. `@EqualsAndHashcode`
+6. `@NoArgsConstructor`
+7. `@RequiredArgsConstructor`
+9. `@AllArgsConstructor`
+
+**BUT**, it should be used with caution, because it can cause performance issues. 
+Consider these articles: 
+1. https://habr.com/ru/company/haulmont/blog/564682/
+2. https://thorben-janssen.com/lombok-hibernate-how-to-avoid-common-pitfalls
+
```
Comment (Nick97-git) [inline java-spring-boot/spring-data-jpa/jv-springboot-data-jpa.md:46]: remove empty line

**Review by Nick97-git — APPROVED** [review]

**Review by hronograf — APPROVED** [review]

**Pull request merged**